<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Framework and workflows for data analysis in the water cycle • StreamFind</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Framework and workflows for data analysis in the water cycle">
<meta name="description" content="The StreamFind R package is a framework for spectrometric and spectroscopic data processing with focus on environmental and quality studies of the water cycle. The StreamFind R package aims to stimulate the use of advanced data analysis (e.g., non-target screening (NTS) and statistical analysis) in routine studies, promoting standardization of data processing and easing the retrospective inclusion of data.">
<meta property="og:description" content="The StreamFind R package is a framework for spectrometric and spectroscopic data processing with focus on environmental and quality studies of the water cycle. The StreamFind R package aims to stimulate the use of advanced data analysis (e.g., non-target screening (NTS) and statistical analysis) in routine studies, promoting standardization of data processing and easing the retrospective inclusion of data.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">StreamFind</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/StreamFind.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/app_guide.html">StreamFind App Guide</a></li>
    <li><a class="dropdown-item" href="articles/demo_ms_raman_fusion_bsa.html">Comparison of Different Bovine Serum Albumin Products</a></li>
    <li><a class="dropdown-item" href="articles/demo_ozonation_project.html">Evaluation of Wastewater Ozonation with Mass Spectrometry</a></li>
    <li><a class="dropdown-item" href="articles/demo_paracetamol_fusion.html">Detection of Quality Deviations in Paracetamol Products</a></li>
    <li><a class="dropdown-item" href="articles/demo_paracetamol_quantification_mcrals.html">Quantification of Active Product Ingredient in a Pharmaceutical</a></li>
    <li><a class="dropdown-item" href="articles/developer_guide.html">StreamFind Developer Guide</a></li>
    <li><a class="dropdown-item" href="articles/general_guide.html">StreamFind General Introduction</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/odea-project/StreamFind/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="streamfind-r-package">StreamFind (R package)<a class="anchor" aria-label="anchor" href="#streamfind-r-package"></a>
</h1></div>
<!-- badges: start -->

<p align="center" width="100%">
<img width="60%" src="reference/figures/logo_StreamFind.png" alt="Logo"></p>
<p>The StreamFind is developed within the project <a href="https://www.bildung-forschung.digital/digitalezukunft/de/wissen/Datenkompetenzen/datenkompetenzen_wissenschaftlichen_nachwuchs/Projekte/stream_find.html" class="external-link">“Flexible data analysis and workflow designer to identify chemicals in the water cycle”</a>, which is funded by the <a href="https://www.bmbf.de" class="external-link">Bundesministerium für Bildung und Forschung (BMBF)</a>. The development is carried out by the <a href="https://www.iuta.de" class="external-link">Institut für Umwelt &amp; Energie, Technik &amp; Analytik e. V. (IUTA)</a>, the <a href="https://www.fzi.de/" class="external-link">Forschungszentrum Informatik (FZI)</a> and supporting partners. The goal of the StreamFind is to develop and assemble data processing workflows for different types of analytical data (e.g., mass spectrometry and spectroscopy) and to apply the workflows in different fields (e.g., environmental and quality studies of the water cycle). StreamFind aims to stimulate the use of advanced data analysis (e.g., non-target screening, statistical analysis, etc.) in routine studies, to promote standardization of data structure and processing, and to facilitate retrospective data evaluation. The StreamFind platform is aimed at scientists, but also at technicians, due to its comprehensive documentation, its well categorized set of integrated modular functions and its embedded graphical user interface.<br><br> The StreamFind development is ongoing, please <a href="mailto:cunha@iuta.de">contact us</a> for questions or collaboration.</p>
<p><img src="reference/figures/logo_group.png" width="100%" style="display: block; margin: auto;"></p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Pre-requisites for the StreamFind are the <a href="https://cran.r-project.org/" class="external-link">R</a> software and the <a href="https://cran.r-project.org/bin/windows/Rtools/" class="external-link">RTools</a> (only applicable for Windows users). RTools is needed for compiling the C++ code used in the StreamFind R package. Assuming that R and RTools are installed, the StreamFind R package can be installed from the GitHub repository via the <a href="https://www.bioconductor.org/install/" class="external-link">BiocManager</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"odea-project/StreamFind"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="docker-setup">Docker Setup<a class="anchor" aria-label="anchor" href="#docker-setup"></a>
</h2>
<p>Build the Docker Image</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>docker build <span class="sc">-</span>t my<span class="sc">-</span>r<span class="sc">-</span>app .</span></code></pre></div>
<p>Start the Container</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>docker run <span class="sc">-</span>it <span class="sc">-</span>p <span class="dv">3838</span><span class="sc">:</span><span class="dv">3838</span> <span class="sc">-</span>p <span class="dv">8787</span><span class="sc">:</span><span class="dv">8787</span> <span class="sc">-</span>v <span class="sc">$</span>(pwd)<span class="sc">:</span><span class="er">/</span>app my<span class="sc">-</span>r<span class="sc">-</span>app</span></code></pre></div>
<p>Once the container is up, you’ll be prompted to select the service you want to run:</p>
<p>Option 1: Run Shiny App Starts the Shiny application, accessible at <a href="http://localhost:3838" class="external-link uri">http://localhost:3838</a>.</p>
<p>Option 2: Run RStudio Server Starts the RStudio Server, accessible at <a href="http://localhost:8787" class="external-link uri">http://localhost:8787</a>.</p>
<p>Default credentials:</p>
<p>Username: rstudio</p>
<p>Password: rstudio</p>
<p>Option 3: Run Both Shiny App and RStudio Server</p>
</div>
<div class="section level2">
<h2 id="other-dependencies">Other dependencies<a class="anchor" aria-label="anchor" href="#other-dependencies"></a>
</h2>
<p>The StreamFind depends on other open source software to process different analytical data. For instance, for non-target screening using mass spectrometry the StreamFind uses the <a href="https://github.com/rickhelmus/patRoon" class="external-link">patRoon</a> R package and its own dependencies. Installation instructions for patRoon and its dependencies can be found <a href="https://rickhelmus.github.io/patRoon/handbook_bd/manual-installation.html#r-prerequisites" class="external-link">here</a>. Consult the documentation for dependencies of other data types.</p>
</div>
<div class="section level2">
<h2 id="suplementary-data">Suplementary data<a class="anchor" aria-label="anchor" href="#suplementary-data"></a>
</h2>
<p>The supplementary <a href="https://github.com/odea-project/StreamFindData" class="external-link">StreamFindData</a> R package holds the data used in examples and other documentation assets of the StreamFind and can also be installed from the GitHub repository.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"odea-project/StreamFindData"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h2>
<p>The documentation and usage examples of the StreamFind R package can be found in the <a href="https://odea-project.github.io/StreamFind/reference/index.html">reference page</a> and <a href="https://odea-project.github.io/StreamFind/articles/index.html">articles</a> of the <a href="https://odea-project.github.io/StreamFind/index.html">webpage</a>, respectively.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>The StreamFind is open source due to public funding and the extensive contribution from scientific literature as well as existing open source software. Below, we reference the research and software that is used within StreamFind. Please note that each open source software or research that StreamFind uses relies on other contributions. Therefore, we recommend to search within each citation for other contributions.<br><br></p>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-xcms03" class="csl-entry">
Benton, H. Paul, Elizabeth J. Want, and Timothy M. D. Ebbels. 2010. “Correction of Mass Calibration Gaps in Liquid Chromatography-Mass Spectrometry Metabolomics Data.” <em>BIOINFORMATICS</em> 26: 2488.
</div>
<div id="ref-proteo02" class="csl-entry">
Chambers, M. C., B. Maclean, R. Burke, D Amodei, D. L. Ruderman, S. Neumann, L. Gatto, et al. 2012a. “A Cross-Platform Toolkit for Mass Spectrometry and Proteomics.” <em>Nature Biotechnology</em> 30 (10): 918–20. <a href="https://doi.org/10.1038/nbt.2377" class="external-link uri">https://doi.org/10.1038/nbt.2377</a>.
</div>
<div id="ref-mzr04" class="csl-entry">
Chambers, Matthew C., Maclean, Brendan, Burke, Robert, Amodei, et al. 2012b. “<span class="nocase">A cross-platform toolkit for mass spectrometry and proteomics</span>.” <em>Nat Biotech</em> 30 (10): 918–20. <a href="https://doi.org/10.1038/nbt.2377" class="external-link uri">https://doi.org/10.1038/nbt.2377</a>.
</div>
<div id="ref-msnbase02" class="csl-entry">
Gatto, Laurent, Sebastian Gibb, and Johannes Rainer. 2020. “MSnbase, Efficient and Elegant r-Based Processing and Visualisation of Raw Mass Spectrometry Data.” <em>bioRxiv</em>.
</div>
<div id="ref-msnbase01" class="csl-entry">
Gatto, Laurent, and Kathryn Lilley. 2012. “MSnbase - an r/Bioconductor Package for Isobaric Tagged Mass Spectrometry Data Visualization, Processing and Quantitation.” <em>Bioinformatics</em> 28: 288–89.
</div>
<div id="ref-patroon01" class="csl-entry">
Helmus, Rick, Thomas L. ter Laak, Annemarie P. van Wezel, Pim de Voogt, and Emma L. Schymanski. 2021. “patRoon: Open Source Software Platform for Environmental Mass Spectrometry Based Non-Target Screening.” <em>Journal of Cheminformatics</em> 13 (1). <a href="https://doi.org/10.1186/s13321-020-00477-w" class="external-link uri">https://doi.org/10.1186/s13321-020-00477-w</a>.
</div>
<div id="ref-patroon02" class="csl-entry">
Helmus, Rick, Bas van de Velde, Andrea M. Brunner, Thomas L. ter Laak, Annemarie P. van Wezel, and Emma L. Schymanski. 2022. “patRoon 2.0: Improved Non-Target Analysis Workflows Including Automated Transformation Product Screening.” <em>Journal of Open Source Software</em> 7 (71): 4029. <a href="https://doi.org/10.21105/joss.04029" class="external-link uri">https://doi.org/10.21105/joss.04029</a>.
</div>
<div id="ref-kpic01" class="csl-entry">
Ji, Hongchao, Fanjuan Zeng, Yamei Xu, Hongmei Lu, and Zhimin Zhang. 2017. “KPIC2: An Effective Framework for Mass Spectrometry-Based Metabolomics Using Pure Ion Chromatograms.” <em>Anal Chem.</em> 14 (89): 7631–40. <a href="https://doi.org/10.1021/acs.analchem.7b01547" class="external-link uri">https://doi.org/10.1021/acs.analchem.7b01547</a>.
</div>
<div id="ref-pugixml01" class="csl-entry">
Kapoulkine, Arseny. 2022. “Pugixml 1.13: Light-Weight, Simple and Fast XML Parser for c++ with XPath Support.” <em>Copyright (C) 2006-2018</em>. <a href="http://pugixml.org" class="external-link uri">http://pugixml.org</a>.
</div>
<div id="ref-mzr02" class="csl-entry">
Keller, Andrew, Jimmy Eng, Ning Zhang, Xiao-jun Li, and Ruedi Aebersold. 2005. “A Uniform Proteomics MS/MS Analysis Platform Utilizing Open XML File Formats.” <em>Mol Syst Biol</em>.
</div>
<div id="ref-proteo01" class="csl-entry">
Kessner, Darren, Matt Chambers, Robert Burke, David Agus, and Parag Mallick. 2008. “ProteoWizard: Open Source Software for Rapid Proteomics Tools Development.” <em>Bioinformatics</em> 24 (21): 2534–36. <a href="https://doi.org/10.1093/bioinformatics/btn323" class="external-link uri">https://doi.org/10.1093/bioinformatics/btn323</a>.
</div>
<div id="ref-mdatools01" class="csl-entry">
Kucheryavskiy, Sergey. 2020. “Mdatools – r Package for Chemometrics.” <em>Chemometrics and Intelligent Laboratory Systems</em> 198: 103937. <a href="https://doi.org/" class="external-link uri">https://doi.org/</a><a href="https://doi.org/10.1016/j.chemolab.2020.103937" class="external-link uri">https://doi.org/10.1016/j.chemolab.2020.103937</a>.
</div>
<div id="ref-camera01" class="csl-entry">
Kuhl, C., R. Tautenhahn, C. Boettcher, T. R. Larson, and S. Neumann. 2012. “CAMERA: An Integrated Strategy for Compound Spectra Extraction and Annotation of Liquid Chromatography/Mass Spectrometry Data Sets.” <em>Analytical Chemistry</em> 84: 283–89. <a href="http://pubs.acs.org/doi/abs/10.1021/ac202450g" class="external-link uri">http://pubs.acs.org/doi/abs/10.1021/ac202450g</a>.
</div>
<div id="ref-mzr03" class="csl-entry">
Martens, Lennart, Matthew Chambers, Marc Sturm, Darren Kessner, Fredrik Levander, Jim Shofstahl, Wilfred H Tang, et al. 2010. “MzML - a Community Standard for Mass Spectrometry Data.” <em>Mol Cell Proteomics</em>. <a href="https://doi.org/10.1074/mcp.R110.000133" class="external-link uri">https://doi.org/10.1074/mcp.R110.000133</a>.
</div>
<div id="ref-genform" class="csl-entry">
Meringer, Markus, Stefan Reinker, Juan Zhang, and Alban Muller. 2011. “MS/MS Data Improves Automated Determination of Molecular Formulas by Mass Spectrometry.” <em>MATCH Commun. Math. Comput. Chem</em> 65 (2): 259–90.
</div>
<div id="ref-mzr01" class="csl-entry">
Pedrioli, Patrick G A, Jimmy K Eng, Robert Hubley, Mathijs Vogelzang, Eric W Deutsch, Brian Raught, Brian Pratt, et al. 2004. “A Common Open Representation of Mass Spectrometry Data and Its Application to Proteomics Research.” <em>Nat Biotechnol</em> 22 (11): 1459–66. <a href="https://doi.org/10.1038/nbt1031" class="external-link uri">https://doi.org/10.1038/nbt1031</a>.
</div>
<div id="ref-qcentroids01" class="csl-entry">
Reuschenbach, Max, Lotta L. Hohrenk-Danzouma, Torsten C. Schmidt, and Gerrit Renner. 2022. “Development of a Scoring Parameter to Characterize Data Quality of Centroids in High-Resolution Mass Spectra.” <em>Analytical and Bioanalytical Chemistry</em> 414 (July): 6635–45. <a href="https://doi.org/10.1007/s00216-022-04224-y" class="external-link uri">https://doi.org/10.1007/s00216-022-04224-y</a>.
</div>
<div id="ref-openms01" class="csl-entry">
Röst, Hannes L., Timo Sachsenberg, Stephan Aiche, Chris Bielow, Hendrik Weisser, Fabian Aicheler, Sandro Andreotti, et al. 2016. “OpenMS: A Flexible Open-Source Software Platform for Mass Spectrometry Data Analysis.” <em>Nature Methods</em> 13 (9): 741–48. <a href="https://doi.org/10.1038/nmeth.3959" class="external-link uri">https://doi.org/10.1038/nmeth.3959</a>.
</div>
<div id="ref-metfrag02" class="csl-entry">
Ruttkies, Christoph, Steffen Neumann, and Stefan Posch. 2019. “Improving MetFrag with Statistical Learning of Fragment Annotations.” <em>BMC Bioinformatics</em> 20 (1): 1–14.
</div>
<div id="ref-metfrag03" class="csl-entry">
Ruttkies, Christoph, Emma L Schymanski, Nadine Strehmel, Juliane Hollender, Steffen Neumann, Antony J Williams, and Martin Krauss. 2019. “Supporting Non-Target Identification by Adding Hydrogen Deuterium Exchange MS/MS Capabilities to MetFrag.” <em>Analytical and Bioanalytical Chemistry</em> 411: 4683–4700.
</div>
<div id="ref-metfrag01" class="csl-entry">
Ruttkies, Christoph, Emma L Schymanski, Sebastian Wolf, Juliane Hollender, and Steffen Neumann. 2016. “MetFrag Relaunched: Incorporating Strategies Beyond in Silico Fragmentation.” <em>Journal of Cheminformatics</em> 8 (1): 1–16.
</div>
<div id="ref-xcms01" class="csl-entry">
Smith, C.A., Want, E.J., O’Maille, G., Abagyan,R., Siuzdak, and G. 2006. “XCMS: Processing Mass Spectrometry Data for Metabolite Profiling Using Nonlinear Peak Alignment, Matching and Identification.” <em>Analytical Chemistry</em> 78: 779–87.
</div>
<div id="ref-xcms02" class="csl-entry">
Tautenhahn, Ralf, Christoph Boettcher, and Steffen Neumann. 2008. “Highly Sensitive Feature Detection for High Resolution LC/MS.” <em>BMC Bioinformatics</em> 9: 504.
</div>
<div id="ref-class01" class="csl-entry">
Venables, W. N., and B. D. Ripley. 2002. <em>Modern Applied Statistics with s</em>. Fourth. New York: Springer. <a href="https://www.stats.ox.ac.uk/pub/MASS4/" class="external-link uri">https://www.stats.ox.ac.uk/pub/MASS4/</a>.
</div>
<div id="ref-mdatools02" class="csl-entry">
Windig, Willem, Neal B. Gallagher, Jeremy M. Shaver, and Barry M. Wise. 2005. “A New Approach for Interactive Self-Modeling Mixture Analysis.” <em>Chemometrics and Intelligent Laboratory Systems</em> 77 (1): 85–96. <a href="https://doi.org/" class="external-link uri">https://doi.org/</a><a href="https://doi.org/10.1016/j.chemolab.2004.06.009" class="external-link uri">https://doi.org/10.1016/j.chemolab.2004.06.009</a>.
</div>
<div id="ref-metfrag04" class="csl-entry">
Wolf, Sebastian, Stephan Schmidt, Matthias Müller-Hannemann, and Steffen Neumann. 2010. “In Silico Fragmentation for Computer Assisted Identification of Metabolite Mass Spectra.” <em>BMC Bioinformatics</em> 11: 1–12.
</div>
<div id="ref-airpls01" class="csl-entry">
Zhang, Zhi-Min, Shan Chen, and Yi-Zeng Liang. 2010. “Baseline Correction Using Adaptive Iteratively Reweighted Penalized Least Squares.” <em>Analyst</em> 135: 1138–46. <a href="https://doi.org/10.1039/B922045C" class="external-link uri">https://doi.org/10.1039/B922045C</a>.
</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/odea-project/StreamFind/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/odea-project/StreamFind/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing StreamFind</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Ricardo Cunha <br><small class="roles"> Maintainer, author </small> <a href="https://orcid.org/0000-0002-6791-0119" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle: experimental"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ricardo Cunha.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
