<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Flexible Workflow Assembler for Advanced Data Analysis • StreamFind</title>
<script src="lightswitch.js"></script><script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Flexible Workflow Assembler for Advanced Data Analysis">
<meta name="description" content="The package StreamFind is a flexible R package designed to support the assembly and execution of standardized workflows for advanced data analysis. Originally developed within the project “Flexible Data Analysis and Workflow Designer to Identify Chemicals in the Water Cycle”, funded by the German Federal Ministry of Education and Research (BMBF), StreamFind offers a modular platform capable of handling diverse data types, including mass spectrometry, spectroscopy, and other structured datasets. The package promotes the use of advanced analysis techniques — such as non-target screening (NTS), statistical modeling, and retrospective data evaluation — in both research and routine quality control workflows. With its standardized approach to data processing, comprehensive documentation, and an integrated graphical user interface, StreamFind is designed to support scientists and technical staff across a wide range of disciplines, from environmental monitoring to pharmaceutical quality assurance.">
<meta property="og:description" content="The package StreamFind is a flexible R package designed to support the assembly and execution of standardized workflows for advanced data analysis. Originally developed within the project “Flexible Data Analysis and Workflow Designer to Identify Chemicals in the Water Cycle”, funded by the German Federal Ministry of Education and Research (BMBF), StreamFind offers a modular platform capable of handling diverse data types, including mass spectrometry, spectroscopy, and other structured datasets. The package promotes the use of advanced analysis techniques — such as non-target screening (NTS), statistical modeling, and retrospective data evaluation — in both research and routine quality control workflows. With its standardized approach to data processing, comprehensive documentation, and an integrated graphical user interface, StreamFind is designed to support scientists and technical staff across a wide range of disciplines, from environmental monitoring to pharmaceutical quality assurance.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">StreamFind</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/StreamFind.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/app_guide.html">StreamFind App Guide</a></li>
    <li><a class="dropdown-item" href="articles/developer_guide.html">StreamFind Developer Guide</a></li>
    <li><a class="dropdown-item" href="articles/general_guide.html">StreamFind General Introduction</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/odea-project/StreamFind/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9">
<p><img src="reference/figures/logos.png" alt="Logo" width="100%" style="display: block; margin: auto;"></p>
<p>StreamFind R package is developed within the project <a href="https://www.bildung-forschung.digital/digitalezukunft/de/bildung/digital-_und_datenkompetenzen/datenkompetenzen_wissenschaftlichen_nachwuchs/Projekte/stream_find.html" class="external-link">“Flexible data analysis and workflow designer to identify chemicals in the water cycle”</a> funded by the <a href="https://www.bmbf.de" class="external-link">German Federal Ministry of Education and Research (BMBF)</a>. The development is carried out by the <a href="https://www.iuta.de" class="external-link">Institut für Umwelt &amp; Energie, Technik &amp; Analytik e. V. (IUTA)</a>, the <a href="https://www.fzi.de/" class="external-link">Forschungszentrum Informatik (FZI)</a> and supporting partners. StreamFind is intended to be a platform for assembling processing workflows for different types of data (e.g. mass spectrometry (MS) and spectroscopy data) with applications in different fields (e.g. environmental studies of the water cycle and quality control of pharmaceuticals). StreamFind aims to stimulate the use of advanced data analysis (e.g. non-target screening, statistical analysis, etc.) in routine studies, to promote standardization of data structure and processing, and to facilitate retrospective data evaluation. The StreamFind platform is aimed at scientists, but also at technicians due to its comprehensive documentation, its well categorized set of integrated modular processing methods and its embedded graphical user interface.</p>
<p>The StreamFind development is ongoing, please <a href="mailto:cunha@iuta.de">contact us</a> for questions or collaboration.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Pre-requisites for the StreamFind are the <a href="https://cran.r-project.org/" class="external-link">R</a> software and the <a href="https://cran.r-project.org/bin/windows/Rtools/" class="external-link">RTools</a> (only applicable for Windows users). RTools is needed for compiling C++ code used in the StreamFind R package. StreamFind also uses python scripts for some of its processing methods, so it is recommended to have the latest <a href="https://www.python.org/downloads/" class="external-link">python</a> installed and added to the environmental variables for Windows users. Assuming that R, RTools and python are installed, the StreamFind R package can be installed from the GitHub repository via the <a href="https://www.bioconductor.org/install/" class="external-link">BiocManager</a>. Note that StreamFind is not yet available on the Bioconductor repository.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"odea-project/StreamFind"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="other-dependencies">Other dependencies<a class="anchor" aria-label="anchor" href="#other-dependencies"></a>
</h2>
<p>The StreamFind depends on other open source software to process different analytic data. For instance, for non-target screening using mass spectrometry the StreamFind uses the <a href="https://github.com/rickhelmus/patRoon" class="external-link">patRoon</a> R package and its own dependencies for some processing methods. When a dependency is not installed, a warning message will be displayed with instructions. Consult the documentation of each processing method for dependencies.</p>
</div>
<div class="section level2">
<h2 id="suplementary-data">Suplementary data<a class="anchor" aria-label="anchor" href="#suplementary-data"></a>
</h2>
<p>The supplementary <a href="https://github.com/odea-project/StreamFindData" class="external-link">StreamFindData</a> R package holds the data used in examples and other documentation assets of the StreamFind and can also be installed from the GitHub repository.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"odea-project/StreamFindData"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="docker-setup">Docker Setup<a class="anchor" aria-label="anchor" href="#docker-setup"></a>
</h2>
<p>The StreamFind can also be used via the <a href="https://www.docker.com/products/docker-desktop/" class="external-link">Docker</a> container. The Docker container is a pre-configured environment with all the necessary dependencies installed. The Docker container can be built and started with the following commands.</p>
<p>Build the Docker container:<br><code>docker build -t my-r-app .</code></p>
<p>Start the Docker container:<br><code>docker run -it -p 3838:3838 -p 8787:8787 -v $(pwd):/app my-r-app</code></p>
<p>Once the container is up, you’ll be prompted to select the service you want to run:</p>
<ul>
<li>
<em>Option 1:</em> Starts the Shiny application, accessible at <a href="http://localhost:3838" class="external-link uri">http://localhost:3838</a>.</li>
<li>
<em>Option 2:</em> Starts the RStudio Server, accessible at <a href="http://localhost:8787" class="external-link uri">http://localhost:8787</a>
<ul>
<li>Default Username is rstudio and Password is rstudio</li>
</ul>
</li>
<li>
<em>Option 3:</em> Starts both the Shiny App and RStudio Server</li>
</ul>
</div>
<div class="section level2">
<h2 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h2>
<p>The documentation and usage examples of the StreamFind R package can be found in the <a href="https://odea-project.github.io/StreamFind/reference/index.html">reference page</a> and <a href="https://odea-project.github.io/StreamFind/articles/index.html">articles</a> of the <a href="https://odea-project.github.io/StreamFind/index.html">webpage</a>, respectively. Note that documentation and articles are under development so not all are yet available.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>The StreamFind is open source due to public funding and the extensive contribution from scientific literature as well as existing open source software. Below, we reference the research and software that is used within StreamFind. Please note that each open source software or research that StreamFind uses relies on other contributions. Therefore, we recommend to search within each citation for other contributions.<br><br></p>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-xcms03" class="csl-entry">
Benton, H. Paul, Elizabeth J. Want, and Timothy M. D. Ebbels. 2010. “Correction of Mass Calibration Gaps in Liquid Chromatography-Mass Spectrometry Metabolomics Data.” <em>BIOINFORMATICS</em> 26: 2488.
</div>
<div id="ref-proteo02" class="csl-entry">
Chambers, M. C., B. Maclean, R. Burke, D Amodei, D. L. Ruderman, S. Neumann, L. Gatto, et al. 2012a. “A Cross-Platform Toolkit for Mass Spectrometry and Proteomics.” <em>Nature Biotechnology</em> 30 (10): 918–20. <a href="https://doi.org/10.1038/nbt.2377" class="external-link uri">https://doi.org/10.1038/nbt.2377</a>.
</div>
<div id="ref-mzr04" class="csl-entry">
Chambers, Matthew C., Maclean, Brendan, Burke, Robert, Amodei, et al. 2012b. “<span class="nocase">A cross-platform toolkit for mass spectrometry and proteomics</span>.” <em>Nat Biotech</em> 30 (10): 918–20. <a href="https://doi.org/10.1038/nbt.2377" class="external-link uri">https://doi.org/10.1038/nbt.2377</a>.
</div>
<div id="ref-msnbase02" class="csl-entry">
Gatto, Laurent, Sebastian Gibb, and Johannes Rainer. 2020. “MSnbase, Efficient and Elegant r-Based Processing and Visualisation of Raw Mass Spectrometry Data.” <em>bioRxiv</em>.
</div>
<div id="ref-msnbase01" class="csl-entry">
Gatto, Laurent, and Kathryn Lilley. 2012. “MSnbase - an r/Bioconductor Package for Isobaric Tagged Mass Spectrometry Data Visualization, Processing and Quantitation.” <em>Bioinformatics</em> 28: 288–89.
</div>
<div id="ref-patroon01" class="csl-entry">
Helmus, Rick, Thomas L. ter Laak, Annemarie P. van Wezel, Pim de Voogt, and Emma L. Schymanski. 2021. “patRoon: Open Source Software Platform for Environmental Mass Spectrometry Based Non-Target Screening.” <em>Journal of Cheminformatics</em> 13 (1). <a href="https://doi.org/10.1186/s13321-020-00477-w" class="external-link uri">https://doi.org/10.1186/s13321-020-00477-w</a>.
</div>
<div id="ref-patroon02" class="csl-entry">
Helmus, Rick, Bas van de Velde, Andrea M. Brunner, Thomas L. ter Laak, Annemarie P. van Wezel, and Emma L. Schymanski. 2022. “patRoon 2.0: Improved Non-Target Analysis Workflows Including Automated Transformation Product Screening.” <em>Journal of Open Source Software</em> 7 (71): 4029. <a href="https://doi.org/10.21105/joss.04029" class="external-link uri">https://doi.org/10.21105/joss.04029</a>.
</div>
<div id="ref-kpic01" class="csl-entry">
Ji, Hongchao, Fanjuan Zeng, Yamei Xu, Hongmei Lu, and Zhimin Zhang. 2017. “KPIC2: An Effective Framework for Mass Spectrometry-Based Metabolomics Using Pure Ion Chromatograms.” <em>Anal Chem.</em> 14 (89): 7631–40. <a href="https://doi.org/10.1021/acs.analchem.7b01547" class="external-link uri">https://doi.org/10.1021/acs.analchem.7b01547</a>.
</div>
<div id="ref-pugixml01" class="csl-entry">
Kapoulkine, Arseny. 2022. “Pugixml 1.13: Light-Weight, Simple and Fast XML Parser for c++ with XPath Support.” <em>Copyright (C) 2006-2018</em>. <a href="http://pugixml.org" class="external-link uri">http://pugixml.org</a>.
</div>
<div id="ref-mzr02" class="csl-entry">
Keller, Andrew, Jimmy Eng, Ning Zhang, Xiao-jun Li, and Ruedi Aebersold. 2005. “A Uniform Proteomics MS/MS Analysis Platform Utilizing Open XML File Formats.” <em>Mol Syst Biol</em>.
</div>
<div id="ref-proteo01" class="csl-entry">
Kessner, Darren, Matt Chambers, Robert Burke, David Agus, and Parag Mallick. 2008. “ProteoWizard: Open Source Software for Rapid Proteomics Tools Development.” <em>Bioinformatics</em> 24 (21): 2534–36. <a href="https://doi.org/10.1093/bioinformatics/btn323" class="external-link uri">https://doi.org/10.1093/bioinformatics/btn323</a>.
</div>
<div id="ref-mdatools01" class="csl-entry">
Kucheryavskiy, Sergey. 2020. “Mdatools – r Package for Chemometrics.” <em>Chemometrics and Intelligent Laboratory Systems</em> 198: 103937. <a href="https://doi.org/" class="external-link uri">https://doi.org/</a><a href="https://doi.org/10.1016/j.chemolab.2020.103937" class="external-link uri">https://doi.org/10.1016/j.chemolab.2020.103937</a>.
</div>
<div id="ref-camera01" class="csl-entry">
Kuhl, C., R. Tautenhahn, C. Boettcher, T. R. Larson, and S. Neumann. 2012. “CAMERA: An Integrated Strategy for Compound Spectra Extraction and Annotation of Liquid Chromatography/Mass Spectrometry Data Sets.” <em>Analytical Chemistry</em> 84: 283–89. <a href="http://pubs.acs.org/doi/abs/10.1021/ac202450g" class="external-link uri">http://pubs.acs.org/doi/abs/10.1021/ac202450g</a>.
</div>
<div id="ref-mzr03" class="csl-entry">
Martens, Lennart, Matthew Chambers, Marc Sturm, Darren Kessner, Fredrik Levander, Jim Shofstahl, Wilfred H Tang, et al. 2010. “MzML - a Community Standard for Mass Spectrometry Data.” <em>Mol Cell Proteomics</em>. <a href="https://doi.org/10.1074/mcp.R110.000133" class="external-link uri">https://doi.org/10.1074/mcp.R110.000133</a>.
</div>
<div id="ref-genform" class="csl-entry">
Meringer, Markus, Stefan Reinker, Juan Zhang, and Alban Muller. 2011. “MS/MS Data Improves Automated Determination of Molecular Formulas by Mass Spectrometry.” <em>MATCH Commun. Math. Comput. Chem</em> 65 (2): 259–90.
</div>
<div id="ref-mzr01" class="csl-entry">
Pedrioli, Patrick G A, Jimmy K Eng, Robert Hubley, Mathijs Vogelzang, Eric W Deutsch, Brian Raught, Brian Pratt, et al. 2004. “A Common Open Representation of Mass Spectrometry Data and Its Application to Proteomics Research.” <em>Nat Biotechnol</em> 22 (11): 1459–66. <a href="https://doi.org/10.1038/nbt1031" class="external-link uri">https://doi.org/10.1038/nbt1031</a>.
</div>
<div id="ref-qcentroids01" class="csl-entry">
Reuschenbach, Max, Lotta L. Hohrenk-Danzouma, Torsten C. Schmidt, and Gerrit Renner. 2022. “Development of a Scoring Parameter to Characterize Data Quality of Centroids in High-Resolution Mass Spectra.” <em>Analytical and Bioanalytical Chemistry</em> 414 (July): 6635–45. <a href="https://doi.org/10.1007/s00216-022-04224-y" class="external-link uri">https://doi.org/10.1007/s00216-022-04224-y</a>.
</div>
<div id="ref-openms01" class="csl-entry">
Röst, Hannes L., Timo Sachsenberg, Stephan Aiche, Chris Bielow, Hendrik Weisser, Fabian Aicheler, Sandro Andreotti, et al. 2016. “OpenMS: A Flexible Open-Source Software Platform for Mass Spectrometry Data Analysis.” <em>Nature Methods</em> 13 (9): 741–48. <a href="https://doi.org/10.1038/nmeth.3959" class="external-link uri">https://doi.org/10.1038/nmeth.3959</a>.
</div>
<div id="ref-metfrag02" class="csl-entry">
Ruttkies, Christoph, Steffen Neumann, and Stefan Posch. 2019. “Improving MetFrag with Statistical Learning of Fragment Annotations.” <em>BMC Bioinformatics</em> 20 (1): 1–14.
</div>
<div id="ref-metfrag03" class="csl-entry">
Ruttkies, Christoph, Emma L Schymanski, Nadine Strehmel, Juliane Hollender, Steffen Neumann, Antony J Williams, and Martin Krauss. 2019. “Supporting Non-Target Identification by Adding Hydrogen Deuterium Exchange MS/MS Capabilities to MetFrag.” <em>Analytical and Bioanalytical Chemistry</em> 411: 4683–4700.
</div>
<div id="ref-metfrag01" class="csl-entry">
Ruttkies, Christoph, Emma L Schymanski, Sebastian Wolf, Juliane Hollender, and Steffen Neumann. 2016. “MetFrag Relaunched: Incorporating Strategies Beyond in Silico Fragmentation.” <em>Journal of Cheminformatics</em> 8 (1): 1–16.
</div>
<div id="ref-orpl01" class="csl-entry">
Sheehy, Guillaume, Fabien Picot, Frédérick Dallaire, Katherine Ember, Tien Nguyen, Kevin Petrecca, Dominique Trudel, and Frédéric Leblond. 2023. “<span class="nocase">Open-sourced Raman spectroscopy data processing package implementing a baseline removal algorithm validated from multiple datasets acquired in human tissue and biofluids</span>.” <em>Journal of Biomedical Optics</em> 28 (2): 025002. <a href="https://doi.org/10.1117/1.JBO.28.2.025002" class="external-link uri">https://doi.org/10.1117/1.JBO.28.2.025002</a>.
</div>
<div id="ref-xcms01" class="csl-entry">
Smith, C.A., Want, E.J., O’Maille, G., Abagyan,R., Siuzdak, and G. 2006. “XCMS: Processing Mass Spectrometry Data for Metabolite Profiling Using Nonlinear Peak Alignment, Matching and Identification.” <em>Analytical Chemistry</em> 78: 779–87.
</div>
<div id="ref-xcms02" class="csl-entry">
Tautenhahn, Ralf, Christoph Boettcher, and Steffen Neumann. 2008. “Highly Sensitive Feature Detection for High Resolution LC/MS.” <em>BMC Bioinformatics</em> 9: 504.
</div>
<div id="ref-tisler01" class="csl-entry">
Tisler, Selina, David I. Pattison, and Jan H. Christensen. 2021. “Correction of Matrix Effects for Reliable Non-Target Screening LC–ESI–MS Analysis of Wastewater.” <em>Analytical Chemistry</em> 93 (24): 8432–41. <a href="https://doi.org/10.1021/acs.analchem.1c00357" class="external-link uri">https://doi.org/10.1021/acs.analchem.1c00357</a>.
</div>
<div id="ref-class01" class="csl-entry">
Venables, W. N., and B. D. Ripley. 2002. <em>Modern Applied Statistics with s</em>. Fourth. New York: Springer. <a href="https://www.stats.ox.ac.uk/pub/MASS4/" class="external-link uri">https://www.stats.ox.ac.uk/pub/MASS4/</a>.
</div>
<div id="ref-mdatools02" class="csl-entry">
Windig, Willem, Neal B. Gallagher, Jeremy M. Shaver, and Barry M. Wise. 2005. “A New Approach for Interactive Self-Modeling Mixture Analysis.” <em>Chemometrics and Intelligent Laboratory Systems</em> 77 (1): 85–96. <a href="https://doi.org/" class="external-link uri">https://doi.org/</a><a href="https://doi.org/10.1016/j.chemolab.2004.06.009" class="external-link uri">https://doi.org/10.1016/j.chemolab.2004.06.009</a>.
</div>
<div id="ref-metfrag04" class="csl-entry">
Wolf, Sebastian, Stephan Schmidt, Matthias Müller-Hannemann, and Steffen Neumann. 2010. “In Silico Fragmentation for Computer Assisted Identification of Metabolite Mass Spectra.” <em>BMC Bioinformatics</em> 11: 1–12.
</div>
<div id="ref-airpls01" class="csl-entry">
Zhang, Zhi-Min, Shan Chen, and Yi-Zeng Liang. 2010. “Baseline Correction Using Adaptive Iteratively Reweighted Penalized Least Squares.” <em>Analyst</em> 135: 1138–46. <a href="https://doi.org/10.1039/B922045C" class="external-link uri">https://doi.org/10.1039/B922045C</a>.
</div>
</div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/odea-project/StreamFind/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/odea-project/StreamFind/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing StreamFind</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Ricardo Cunha <br><small class="roles"> Maintainer, author, funder </small> <a href="https://orcid.org/0000-0002-6791-0119" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle: experimental"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ricardo Cunha.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
